src/components/tools/geometry-tool/line-dialog.tsx

```diff
@@ -0,0 +1,147 @@
+import { getBoundingBoxIntersections, solveForY } from "../../../models/tools/geometry/jxg-movable-line";
+import { round } from "lodash";
+
+interface IProps {
+  line: JXG.Line;
+  isOpen: boolean;
+  onAccept: (line: JXG.Line, point1: [number, number], point2: [number, number]) => void;
+  onClose: () => void;
+}
+
+interface IState {
+  slope: string;
+  intercept: string;
+}
+
+export default
+class LineDialog extends React.Component<IProps, IState> {
+  public state = {
+            slope: "" + round(this.props.line.getSlope(), 1) || "",
+            intercept: "" + round(this.props.line.getRise(), 1) || "",
+          };
+
+  public render() {
+    const errorMessage = this.getValidationError();
+    return (
+      <Dialog
+        icon="text-highlight"
+        isOpen={this.props.isOpen}
+        onClose={this.props.onClose}
+        title={`Edit Line`}
+        canOutsideClickClose={false}
+      >
+        <div className="nc-attribute-name-prompt">Slope:</div>
+        <input
+          className="nc-attribute-name-input pt-input"
+          type="text"
+          maxLength={5}
+          value={this.state.slope}
+          onChange={this.handleSlopeChange}
+          onKeyDown={this.handleKeyDown}
+          dir="auto"
+        />
+        <div className="nc-attribute-name-prompt">y-intercept:</div>
+        <input
+          className="nc-attribute-name-input pt-input"
+          type="text"
+          maxLength={5}
+          value={this.state.intercept}
+          onChange={this.handleInterceptChange}
+          onKeyDown={this.handleKeyDown}
+          dir="auto"
+        />
+        <div className="nc-dialog-error">
+          {errorMessage}
+        </div>
+        <div className="nc-dialog-buttons">
+          <Button
+            className="nc-dialog-button pt-intent-primary"
+            text="OK"
+            onClick={this.handleAccept}
+            disabled={errorMessage != null}
+          />
+          <Button className="nc-dialog-button" text="Cancel"  onClick={this.handleCancel}/>
+        </div>
+      </Dialog>
+    );
+  }
+
+  private handleSlopeChange = (evt: React.FormEvent<HTMLInputElement>) => {
+    this.setState({slope: (evt.target as HTMLInputElement).value });
+  }
+
+  private handleInterceptChange = (evt: React.FormEvent<HTMLInputElement>) => {
+    this.setState({intercept: (evt.target as HTMLInputElement).value });
+  }
+
+  private getValidationError = () => {
+    const { line } = this.props;
+    const slope = parseFloat(this.state.slope);
+    const intercept = parseFloat(this.state.intercept);
+    if (isNaN(slope) || isNaN(intercept)) {
+      return "Please enter a valid integer or decimal for slope and intercept.";
+    }
+    const intersections = getBoundingBoxIntersections(slope, intercept, line.board);
+    if (intersections.length < 2) {
+      return "Line would fall offscreen; please edit the slope or intercept.";
+    }
+  }
+
+  private getLineControlPoints = () => {
```

Comment:
I'd like to test some of this logic too, but I couldn't think of a way to do it without pulling these functions out. Maybe that's the only thing to do...